wizeline/project-lab
# Feat - add impersonation page for admin users

### What does this PR do?

I added a page `/impersonate` which allows admins to impersonate other users from the app

#### Where should the reviewer start?

Prerequisites:
- The admin must be an admin user
- The user that will be impersonated must exist in the `User` table

After this, the reviewer should have access to `/impersonate` page.

#### How should this be manually tested?

- Pull the changes and run the project locally
- Set yourself as an `ADMIN` User in the `User` table.
- Add a different User in the `User` table, you can do this by going to the `manager` page -> Admin users and add a new user.
- Go to the `/impersonate` URL, I didn't add a direct button as this is a critical feature that shouldn't be accessed by all Admins.
- Type the email from the user you just created and click 'Switch user', then click 'GO HOME' button
- You should appear as the impersonated User.
- To stop impersonating, go to the `/impersonate` URL and click 'STOP IMPERSONATING', then 'GO HOME', you should now be your current user.

#### Screenshots

<img width="1428" alt="image" src="https://user-images.githubusercontent.com/96010361/161979400-74d8b84f-e5b1-4c13-9f0e-890b0972d4fb.png">

<img width="1277" alt="image" src="https://user-images.githubusercontent.com/96010361/161979674-db1552d8-3ca9-4c1e-aa8d-daa45b823264.png">

![image](https://user-images.githubusercontent.com/96010361/161979830-224bab7a-025e-4451-890f-0c714e9aea93.png)
